core/src/main/scala/org/apache/spark/util/AkkaUtils.scala

```diff
@@ -30,6 +34,61 @@ import org.apache.log4j.{Level, Logger}
 import org.apache.spark.{Logging, SecurityManager, SparkConf, SparkEnv, SparkException}
 
 /**
+ * Associates a timeout with a configuration property so that a TimeoutException can be
+ * traced back to the controlling property.  The main constructor takes a generic timeout
+ * and description while the auxilary constructor uses a specific property defined in the
+ * given configuration.
+ * @param duration timeout duration in milliseconds
+ * @param description description to be displayed in a timeout exception
+ */
+class ConfiguredTimeout(duration: FiniteDuration, description: String) {
```

Comment:
this need to be scoped at least to `private[spark]`, possibly even tighter.  Same goes for other classes & methods.  We can wait till the rest of the design shakes out a little more before worrying about that too much, though